extensions/blocks/story/edit.js

```diff
@@ -0,0 +1,215 @@
+	getImageSizeOptions() {
+		const { imageSizes } = this.props;
+		return map( imageSizes, ( { name, slug } ) => ( { value: slug, label: name } ) );
+	}
+	updateMediaFilesSize = sizeSlug => {
+		const { mediaFiles } = this.props.attributes;
+		const { resizedMediaFiles } = this.props;
+
+		const updatedMediaFiles = mediaFiles.map( image => {
+			const resizedImage = resizedMediaFiles.find(
+				( { id } ) => parseInt( id, 10 ) === parseInt( image.id, 10 )
+			);
+			const url = get( resizedImage, [ 'sizes', sizeSlug, 'source_url' ] );
+			return {
+				...image,
+				...( url && { url } ),
+			};
+		} );
+
+		this.setAttributes( { mediaFiles: updatedMediaFiles, sizeSlug } );
+	};
+	render() {
+		const { attributes, className, isSelected, noticeOperations, noticeUI } = this.props;
+		const { align, mediaFiles } = attributes;
+
+		const imageSizeOptions = this.getImageSizeOptions();
+		const controls = (
+			<Controls
+				allowedMediaTypes={ ALLOWED_MEDIA_TYPES }
+				attributes={ attributes }
+				imageSizeOptions={ imageSizeOptions }
+				onChangeImageSize={ this.updateMediaFilesSize }
+				onSelectMedia={ this.onSelectMedia }
+				setAttributes={ attrs => this.setAttributes( attrs ) }
+			/>
+		);
+
+		if ( mediaFiles.length === 0 ) {
+			return (
+				<Fragment>
+					{ controls }
+					<MediaPlaceholder
+						icon={ <BlockIcon icon={ icon } /> }
+						className={ className }
+						labels={ {
+							title: __( 'Story', 'jetpack' ),
+							instructions: __(
+								'Drag images and videos, upload new ones or select files from your library.',
+								'jetpack'
+							),
+						} }
+						onSelect={ this.onSelectMedia }
+						accept={ ALLOWED_MEDIA_TYPES.map( type => type + '/*' ).join( ',' ) }
+						allowedTypes={ ALLOWED_MEDIA_TYPES }
+						multiple
+						notices={ noticeUI }
+						onError={ noticeOperations.createErrorNotice }
+					/>
+				</Fragment>
+			);
+		}
+		return (
+			<Fragment>
+				{ controls }
+				{ noticeUI }
+				<Story
+					align={ align }
+					className={ className }
+					mediaFiles={ mediaFiles }
+					onError={ noticeOperations.createErrorNotice }
+				/>
+				<DropZone onFilesDrop={ this.addFiles } />
+				{ isSelected && (
+					<div className="wp-block-jetpack-story__add-item">
+						<FormFileUpload
+							multiple
+							isLarge
+							className="wp-block-jetpack-story__add-item-button"
+							onChange={ this.uploadFromFiles }
+							accept={ ALLOWED_MEDIA_TYPES.map( type => type + '/*' ).join( ',' ) }
+							icon="insert"
+						>
+							{ __( 'Upload a media', 'jetpack' ) }
+						</FormFileUpload>
+					</div>
+				) }
+			</Fragment>
+		);
+	}
+}
+export default compose(
+	withSelect( ( select, props ) => {
+		const imageSizes = select( 'core/editor' ).getEditorSettings().imageSizes;
+		const resizedMediaFiles = props.attributes.ids.reduce( ( currentResizedMediaFiles, id ) => {
+			const image = select( 'core' ).getMedia( id );
+			const sizes = get( image, [ 'media_details', 'sizes' ] );
```

Comment:
You could replace `get` with `image?.media_details?.sizes` if you're not sure all those steps exist.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Indeed that looks better 👍 

Btw, this file is heavily inspired by `slideshow/edit.js` and I'm still not sure what will keep, so a lot of code might go actually